src/platform/silabs/wifi/wf200/WifiInterface.cpp

```diff
@@ -293,6 +293,17 @@ sl_status_t get_all_counters(void)
     return result;
 }
 
+/**
+ * @brief Convert RCPI to RSSI
+ * @param[in]  rcpi: Received Channel Power Indicator value
+ * @return RSSI value
+ */
+int16_t ConvertRcpiToRssi(uint32_t rcpi) {
+    // Checking for the overflows
+    // If rcpi is greater than or equal to INT16_MAX, return INT16_MIN to indicate an error
```

Comment:
Even if in theory this should never happen, we are not interested of checking if the input if bigger than a int16. We need to check that the output is.

In others, do the math with an int32 (need to check if 64 isn't better here), if the result is bigguer than a int16 return an error.

Reply:
added limit checks for the return